wagtail/images/models.py

```diff
@@ -236,6 +236,17 @@ def get_file_hash(self):
 
         return self.file_hash
 
+    def _set_image_file_metadata(self):
+        self.file.open()
+
+        # Set new image file size
+        self.file_size = self.file.size
+
+        # Set new image file hash
+        self.file.seek(0)
```

Comment:
According to https://docs.djangoproject.com/en/4.0/ref/files/file/#django.core.files.File.open, this also does `seek(0)`, so this feels extra?

The other thing that keeps nagging at me: do we need to close the file, especially when calling this outside of the form save context?

Reply:
@zerolab Now removed in 0d4a52644850432c51c1e7f94651f58b7d5ccb4a. I'm very cautious about introducing changes to the file operations here, especially as we might be dealing with custom storage backends that don't strictly follow Django's spec, but you're right, the seek(0) does indeed happen during file.open, even if the file is already opened - and as far as I can see this behaviour happens outside the logic that would be overridden for storage backends like S3, so we can reasonably rely on it.

I'm pretty sure I remember running into "file is not opened for reading" operations a long time back when generating renditions and the like, so I'd rather not introduce any new 'close' operations unless there's evidence that we need them.

Reply:
> I'm pretty sure I remember running into "file is not opened for reading" operations a long time back when generating renditions and the like, so I'd rather not introduce any new 'close' operations unless there's evidence that we need them.

Sounds good to me. :shipit: